client_plugin/Makefile

```diff
@@ -281,6 +288,29 @@ CLEANESX_SH   := $(DEPLOY_TOOLS) cleanesx
 deploy-esx: clean-esx
 	$(DEPLOY_ESX_SH) "$(ESX)" "$(VIB_BIN)"
 
+test-upgrade:
+	$(BUILD) upgrade-test
+upgrade-test:
+	@echo "Upgrade test: from ver $(UPGRADE_FROM_VER) to ver $(UPGRADE_TO_VER)"
+	@echo "Upgrade test step 1: deploy on $(ESX) with $(FROM_VIB_URL)"
+	$(DEPLOY_ESX_FOR_UPGRADE)  "$(ESX)" "$(FROM_VIB_URL)"
+	@echo "Upgrade test step 2.1: remove plugin $(MANAGED_PLUGIN_NAME) on $(VM1)"
+	$(CLEANVM_SH) "$(VM1)" "$(MANAGED_PLUGIN_NAME)"
+	@echo "Upgrade test step 2.2: deploy plugin vmware/docker-volume-vsphere:$(UPGRADE_FROM_VER) on $(VM1)"
+	$(DEPLOY_VM_SH) "$(VM1)" vmware/docker-volume-vsphere:$(UPGRADE_FROM_VER)
+	$(SSH) $(VM1) "systemctl restart docker || service docker restart"
+	@echo "Upgrade test step 3: run pre-upgrade test"
+	$(GO) test -v -timeout 30m -tags runpreupgrade $(E2E_Tests)
+	@echo "Upgrade test step 4: deploy on $(ESX) with $(TO_VIB_URL)"
+	$(DEPLOY_ESX_FOR_UPGRADE)  "$(ESX)" "$(TO_VIB_URL)"
+	@echo "Upgrade test step 5.1: remove plugin $(MANAGED_PLUGIN_NAME) on $(VM1)"
+	$(CLEANVM_SH) "$(VM1)" "$(MANAGED_PLUGIN_NAME)"
+	@echo "Upgrade test step 5.2: deploy plugin vmware/docker-volume-vsphere:$(UPGRADE_TO_VER) on $(VM1)"
+	$(DEPLOY_VM_SH) "$(VM1)" vmware/docker-volume-vsphere:$(UPGRADE_TO_VER)
+	$(SSH) $(VM1) "systemctl restart docker || service docker restart"
```

Comment:
Without restart, I hit this error when creating a volume http://vmware.github.io/docker-volume-vsphere/documentation/faq.html

# docker volume create -d vsphere vol5
Error response from daemon: create vol5: Post http://%2Frun%2Fdocker%2Fplugins%2Fvsphere.sock/VolumeDriver.Create: dial unix /run/docker/plugins/vsphere.sock: connect: no such file or directory